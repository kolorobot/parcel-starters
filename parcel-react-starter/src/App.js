import React, { useEffect, useState } from 'react';

export default function App() {

    const [count, setCount] = useState(0);
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        setIsLoading(true);
        setTimeout(() => {
            setCount(1);
            setIsLoading(false);
        }, Math.floor(Math.random() * 1000));
    }, []);

    return (
        <button data-testid='count-button' disabled={isLoading} onClick={() => setCount(count + 1)}>{count}</button>
    );
}
