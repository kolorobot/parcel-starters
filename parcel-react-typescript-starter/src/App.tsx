import * as React from 'react';
import { useEffect, useState } from 'react';

type AppProps = {
    onCounterUpdate?(value: number): void;
}

const App: React.FC<AppProps> = ({ onCounterUpdate }) => {

    const [count, setCount] = useState(0);
    const [isLoading, setIsLoading] = useState(false);

    useEffect(() => {
        setIsLoading(true);
        setTimeout(() => {
            setCount(1);
            setIsLoading(false);
        }, Math.floor(Math.random() * 1000));
    }, []);

    const handleClick = () => {
        setCount(count + 1);
        if (onCounterUpdate) {
            onCounterUpdate(count);
        }
    };

    return (
        <button data-testid='count-button' disabled={isLoading} onClick={handleClick}>{count}</button>
    );
};

export default App;
