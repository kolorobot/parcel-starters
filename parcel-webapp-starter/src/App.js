export default function App() {
    const div = document.createElement('div');
    div.innerHTML = `
    <label for='text'>Enter some text</label>
    <input id='text' />
    <button>Print</button>`;

    const button = div.querySelector('button');
    const input = div.querySelector('input');

    button.addEventListener('click', () => {
        setTimeout(() => {
            const resultContainer = document.createElement('div');
            resultContainer.innerHTML = `<div data-testid='printed-text'>${input.value}</div>`;
            div.appendChild(resultContainer);
        }, Math.floor(Math.random() * 200));
    });
    return div;
}
