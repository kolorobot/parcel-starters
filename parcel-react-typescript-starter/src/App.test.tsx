import * as React from 'react';
import App from './App';
import { getByTestId, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('App', () => {
    it('waits for the button to be enabled to click it', async () => {
        const { container } = render(<App />);
        const getCountButton = () => getByTestId(container, 'count-button');

        await waitFor(
            () => expect(getCountButton()).not.toHaveAttribute('disabled'),
        );

        userEvent.click(getCountButton());
        expect(container).toMatchSnapshot();
    });
});
