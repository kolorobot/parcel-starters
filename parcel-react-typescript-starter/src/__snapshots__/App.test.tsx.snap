// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App It waits for the button to be enabled and clicks it 1`] = `
<div>
  <button
    data-testid="count-button"
  >
    2
  </button>
</div>
`;

exports[`App waits for the button to be enabled and clicks it 1`] = `
<div>
  <button
    data-testid="count-button"
  >
    2
  </button>
</div>
`;

exports[`App waits for the button to be enabled to click it 1`] = `
<div>
  <button
    data-testid="count-button"
  >
    2
  </button>
</div>
`;
